pytorch_pfn_extras/training/triggers/function_trigger.py

```diff
@@ -0,0 +1,54 @@
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Mapping,
+    Optional,
+    Sequence,
+)
+
+from pytorch_pfn_extras.training import trigger as trigger_module
+from pytorch_pfn_extras.training._manager_protocol import (
+    ExtensionsManagerProtocol,
+)
+
+if TYPE_CHECKING:
+    from pytorch_pfn_extras.training._trigger_util import TriggerLike
+
+
+class FunctionTrigger(trigger_module.Trigger):
+    def __init__(
+        self,
+        fn: Callable[..., bool],
+        args: Optional[Sequence[Any]] = None,
+        kwargs: Optional[Mapping[str, Any]] = None,
+        trigger: "TriggerLike" = (1, "iteration"),
+    ) -> None:
+        self._fn = fn
+        self._args = args or []
+        self._kwargs = kwargs or {}
+        self._interval_trigger = trigger_module.get_trigger(trigger)
+
+    def __call__(self, manager: ExtensionsManagerProtocol) -> bool:
+        if not self._interval_trigger(manager):
+            return False
+
+        return self._fn(*self._args, **self._kwargs)
```

Comment:
I wonder if it would be good to add something like the current iteration/epoch as positional arguments

Reply:
If iteration or epoch information is needed, it is assumed that users will implement a function that explicitly takes `manager` as an argument, and pass `manager` to `args`/`kwargs` to achieve this.